src/config.rs

```diff
@@ -211,22 +212,24 @@ pub fn construct_config(pc: ProxyConfig) -> Result<Config, Error> {
         Some(parse_default(CA_ADDRESS, default_istiod_address)?)
     });
 
-    let xds_root_cert_provider = parse_default(
-        XDS_ROOT_CERT_PROVIDER_ENV,
-        DEFAULT_ROOT_CERT_PROVIDER.to_string(),
-    )?;
+    let xds_root_cert_provider =
+        parse_default(XDS_ROOT_CA_ENV, DEFAULT_ROOT_CERT_PROVIDER.to_string())?;
     let xds_root_cert = if Path::new(&xds_root_cert_provider).exists() {
-        RootCert::File(xds_root_cert_provider.parse().unwrap())
+        RootCert::File(xds_root_cert_provider.into())
+    } else if xds_root_cert_provider.eq(&CERT_SYSTEM.to_string()) {
```

Comment:
nit: technically this should probably be first in case they have a file named "system"

Reply:
also I think you can jsut use `==` instead of `.eq` but not a big deal